### DISCUSSION TITLE
A systematic approach to hotkey prompts

### DISCUSSION THREAD

**Discussion by x9136** [Chatterino/chatterino2#4627] [Ideas]
Hello!

Chatterino has many hidden but useful options that you can only learn about if you want to learn the app or google it.

One way to simplify learning is to show the assigned hotkey next to the option.
Search → Search [⌘F]
Open player in browser → Open player in browser [⌘P]
Copy message → Copy message [Shift+RMB]

You can assign hotkeys to options in the same format in any part of the application. If done correctly, users rarely have to find out if the hotkey they want is on the list and what it is now. An option with the hint for example:
**[  ] Show message reply button**
[Shift+RMB] on a nickname for reply.

I also suggest extending the default list of hotkeys to reduce the need to configure them. Candidates for default hotkeys are popular commands that currently have no hotkeys. For example: Open player in browser.

What do you think about it?

**Comment by pajlada** [comment]
Yeah seems good, we can add the menu hint in more places

**Comment by Mm2PL** [comment]
Is the Qt keybinding display thing broken on mac? We've had a bunch of menu hints for quite a while (since [#4039](https://github.com/Chatterino/chatterino2/pull/4045), oct 2022).
![image](https://github.com/Chatterino/chatterino2/assets/25011746/6993dd22-4039-45de-b879-13534eff31f4)

Although we are still missing them in places. Especially mouse based bindings, as those aren't covered by the Hotkey system.

**Reply by x9136** [reply]
> Is the Qt keybinding display thing broken on mac?
<img width="204" alt="Screenshot 2023-05-27 at 18 38 59" src="https://github.com/Chatterino/chatterino2/assets/60105060/2f73276f-db45-4b36-b1b4-a1c4ce1e1958">

Chatterino 2.4.4 (commit [29a14627](https://github.com/Chatterino/chatterino2/commit/29a14627)) built with Qt 6.5.0
Running on macOS Ventura (13.2), kernel: 22.3.0

**Reply by Mm2PL** [reply]
Very interesting. I'll split off an issue about this not working. https://github.com/Chatterino/chatterino2/issues/4657

**Comment by x9136** [comment]
Closed due no updates.
